sdk/evm/evm.go

```diff
@@ -676,14 +676,27 @@ func (client Client) CreateDummyBlocks(
 	}
 	gasPrice := big.NewInt(legacy.BaseFee)
 	txSigner := types.LatestSignerForChainID(chainID)
+	blockNumber, err := client.BlockNumber()
+	if err != nil {
+		return fmt.Errorf("client.BlockNumber failure: %w", err)
+	}
+	nonce, err := client.NonceAt(addr.Hex())
+	if err != nil {
+		return fmt.Errorf("client.NonceAt failure: %w", err)
+	}
 	for i := 0; i < numBlocks; i++ {
-		prevBlockNumber, err := client.BlockNumber()
-		if err != nil {
+		// it may be the case that we hit an outdated node with the rpc, so lets not fully trust the API
+		if blockNumberFromAPI, err := client.BlockNumber(); err != nil {
 			return fmt.Errorf("client.BlockNumber failure at step %d: %w", i, err)
+		} else if blockNumberFromAPI > blockNumber {
+			// changes from outside
+			blockNumber = blockNumberFromAPI
```

Comment:
why not we always use blockNumberFromAPI instead?

Reply:
because API can be outdated, and we have locally better information in that case. we now a new tx is already accepted.